src/main/kotlin/com/hj/leetcode/kotlin/problem2192/Solution.kt

```diff
@@ -0,0 +1,49 @@
+package com.hj.leetcode.kotlin.problem2192
+
+/**
+ * LeetCode page: [2192. All Ancestors of a Node in a Directed Acyclic Graph](https://leetcode.com/problems/all-ancestors-of-a-node-in-a-directed-acyclic-graph/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(nE+n^2*log(n)) and Space O(n^2+E) where E is the size of edges;
+     */
+    fun getAncestors(n: Int, edges: Array<IntArray>): List<List<Int>> {
+        val adjacencyList = adjacencyList(n, edges)
+        val inDegree = inDegree(n, edges)
+        val ancestors = List(n) { mutableSetOf<Int>() }
+        val zeroDegreeNodes = ArrayDeque<Int>()
+
+        (0..<n).filterTo(zeroDegreeNodes) { inDegree[it] == 0 }
+        while (zeroDegreeNodes.isNotEmpty()) {
+            repeat(zeroDegreeNodes.size) {
```

Comment:
repeat is unnecessary

Reply:
Resolved by https://github.com/hj-core/hj-leetcode-kotlin/pull/389/commits/a6d109d2c72e849542b38ea3941f99cef53fa0e4